src/main/java/com/threem/carrental/app/service/mapper/CarMapper.java

```diff
@@ -0,0 +1,71 @@
+package com.threem.carrental.app.service.mapper;
+
+import com.threem.carrental.app.model.dto.CarDto;
+import com.threem.carrental.app.model.entity.BranchEntity;
+import com.threem.carrental.app.model.entity.CarEntity;
+import org.springframework.stereotype.Service;
+
+/**
+ * @author Marika Grzebieniowska on 27.05.2018
+ * @project car-rental-app
+ */
+
+@Service
+public class CarMapper {
+
+    public CarEntity toCarEntity(CarDto fromCarDto) {
+        BranchEntity dummyBranchEntity = new BranchEntity();
+        if (fromCarDto.getBranchId() != null) {
+            dummyBranchEntity.setId(fromCarDto.getBranchId());
+        }
+
+        CarEntity carEntity = new CarEntity().builder()
+                .id(fromCarDto.getCarId())
+                .vin(fromCarDto.getVin())
+                .make(fromCarDto.getMake())
+                .model(fromCarDto.getModel())
+                .bodyType(fromCarDto.getBodyType())
+                .year(fromCarDto.getYear())
+                .colour(fromCarDto.getColour())
+                .mileage(fromCarDto.getMileage())
+                .status(fromCarDto.getStatus())
+                .dailyRate(fromCarDto.getDailyRate())
+                .engineType(fromCarDto.getEngineType())
+                .engineCapacity(fromCarDto.getEngineCapacity())
+                .segment(fromCarDto.getSegment())
+                .transmission(fromCarDto.getTransmission())
+                .seats(fromCarDto.getSeats())
+                .doors(fromCarDto.getDoors())
+                .branch(dummyBranchEntity) // dummy branch with only branchId
```

Comment:
remember when you will be develop updating of Car in service making branch will all fields